lua/entities/starfall_processor/shared.lua

```diff
@@ -143,6 +230,10 @@ local function MenuOpen( ContextMenu, Option, Entity, Trace )
 	end
 end
 
+function getCodebaseChecksum(files)
+	return tostring(math.random()):sub(-6)
+end
+
```

Comment:
That is a really bad idea, file time is unique per file because it only increases, random may generate way too many false positives.

Reply:
The problem was that non saved new files wouldn't have any file time, or any files that have changed without save (unsure on this one) wouldn't have any different file time. Things not really a checksum anymore just a unique id, the chances of a same id should be extremely low.

The cache is really only valid for the latest used codebase and none of the other previous

Reply:
Just wanna clarify that this cache is only for spawning in new chips. Updating chips just updates the files in place stored in `self.files`